### PR TITLE
[MM-36420] Add flags and logic to control mysql and minio operator deployments in cloud clusters.

### PR DESCRIPTION
**Pull request by stylianosrigas**
#### Summary
- Add flags and logic to control mysql and minio operator deployments in cloud clusters.

#### Ticket Link
- https://mattermost.atlassian.net/browse/MM-36420

#### Release Note
```release-note
- Add flags and logic to control mysql and minio operator deployments in cloud clusters.
```

### CODE REVIEW

**Review by gabrieljackson — APPROVED** [review]
LGTM!

**Review by Szymongib — APPROVED** [review]
LGTM!

File: model/installation_request.go [hunk 1]
```diff
@@ -112,6 +124,13 @@ func (request *CreateInstallationRequest) Validate() error {
 			return errors.Wrap(err, "single tenant database config is invalid")
 		}
 	}
+
+	if !deployMinioOperator && request.Filestore == "minio-operator" {
+		return errors.Errorf("minio filestore cannot be used when minio operator is not deployed")
+	}
+	if !deployMySQLOperator && request.Database == "mysql-operator" {
+		return errors.Errorf("mysql operator database cannot be used when mysql operator is not deployed")
```
Comment (angeloskyratzakos) [inline model/installation_request.go:132]: minor: `Errorf` could be `New`

**Review by angeloskyratzakos — APPROVED** [review]
Nice one!

**Pull request merged**